### ISSUE TITLE
Auto-register Vue components

### ISSUE THREAD

**Issue by jakedohm** [besteadfast/carbon#46]
## TLDR
Components in src/js/components should be auto-registered into Vue, so you don't have to manually import each component.

## Spec
- Components in src/js/components **and all subfolders** will be automatically registered
- Component name will match the filename. `Button.vue` becomes `<Button>`
  - **Note: this will require the developer to make sure they don't name two components exactly the same, which I'm fine with**.
- Lazy loading: If some components aren't required on page-load, we should be able to lazy load them. To lazy load a component, you name the file `ComponentName.lazy.vue`

_This also will take care of #25._
